commit ac1c2f259b44d1229a65a3e639b09d57a4e2a53b
Author: Arne Schwabe <arne@rfc2549.org>
Date:   Fri Mar 21 14:18:35 2014 +0100

    Move the initialization of the environment to the top so c2.es is initialized
    
    Acked-by: Gert Doering <gert@greenie.muc.de>
    Message-Id: <1395407925-25518-3-git-send-email-arne@rfc2549.org>
    URL: http://article.gmane.org/gmane.network.openvpn.devel/8372
    
    Signed-off-by: Gert Doering <gert@greenie.muc.de>

diff --git a/src/openvpn/init.c b/src/openvpn/init.c
index 3b72b96..c50b5c2 100644
--- a/src/openvpn/init.c
+++ b/src/openvpn/init.c
@@ -3339,6 +3339,10 @@ init_instance (struct context *c, const struct env_set *env, const unsigned int
   /* init garbage collection level */
   gc_init (&c->c2.gc);
 
+  /* inherit environmental variables */
+  if (env)
+     do_inherit_env (c, env);
+
   /* signals caught here will abort */
   c->sig->signal_received = 0;
   c->sig->signal_text = NULL;
@@ -3390,10 +3394,6 @@ init_instance (struct context *c, const struct env_set *env, const unsigned int
   if (c->mode == CM_P2P || c->mode == CM_TOP)
     do_option_warnings (c);
 
-  /* inherit environmental variables */
-  if (env)
-    do_inherit_env (c, env);
-
 #ifdef ENABLE_PLUGIN
   /* initialize plugins */
   if (c->mode == CM_P2P || c->mode == CM_TOP)
